server/cpho/views/indicators.py

```diff
@@ -95,3 +118,259 @@ def get_context_data(self, **kwargs):
+        try:
+            csv_file = self.request.FILES["csv_file"]
+            if not csv_file.name.endswith(".csv"):
+                messages.error(self.request, tdt("File is not CSV type"))
+                raise forms.add_error(tdt("File is not CSV type"))
+                return HttpResponseRedirect(reverse("upload_indicator"))
+            if csv_file.multiple_chunks():
+                messages.error(
+                    self.request,
+                    tdt(
+                        "Uploaded file is too big. Maximum size allowed is 2 MB"
+                    ),
+                )
+                raise forms.ValidationError(
+                    tdt(
+                        "Uploaded file is too big. Maximum size allowed is 2 MB"
+                    )
+                )
+                return HttpResponseRedirect(reverse("upload_indicator"))
+
+            # ### Please dont remove this code; might need for custom parsing later
+
+            # file_data = csv_file.read().decode("utf-8-sig")
+            # lines = file_data.split("\n")
+            # data_dict = []
+            # for idx, line in enumerate(lines):
+            #     # print(line)
+            #     fields = line.split(",")
+            #     if idx == 0:
+            #         headers = [s.strip() for s in fields]
+            #         # TODO: check if headers are as expected
+            #     else:
+            #         if len(fields) != len(headers):
+            #             continue
+            #         data_row = {}
+            #         for i, field in enumerate(fields):
+            #             data_row[headers[i]] = field.strip()
+            #         data_dimension = self.deduce_dimension_type(data_row, idx)
+            #         print(data_dimension)
+            #         data_dict.append(data_row)
+            #         # print("SUCCESSFULLY ADDED ROW: ", idx)
+            # # print(data_dict)
+
+            file_data = csv_file.read().decode("utf-8-sig").splitlines()
+            # Not sure if this is needed in the future
+            # file_data = [
+            #     unicodedata.normalize("NFKD", line) for line in file_data
+            # ]
+            reader = csv.DictReader(file_data)
+            data_dict = []
+            for idx, data_row in enumerate(reader):
+                # Very annoying that i have to do this; but gives me errors otherwise
+                for key, value in data_row.items():
+                    data_row[key] = value.strip()
+                data_dimension = self.deduce_dimension_type(data_row, idx)
+                print(data_dimension)
+                data_dict.append(data_row)
+
+            messages.success(self.request, tdt("Data Uploaded Successfully"))
+
+        except Exception as err:
+            messages.error(
+                self.request, tdt("Unable to upload file: ") + repr(err)
+            )
+
+    def deduce_dimension_type(self, record, csv_line_number):
```

Comment:
even this can go in the form I guess? Ideally the form has all the logic, the view is just there to call form.save, message and redirect